Makefile

```diff
@@ -0,0 +1,18 @@
+CHANNEL ?= unpublished
+CHARM := sriov-cni
+
+setup-env:
+	bash script/bootstrap
+
+charm: setup-env
+	bash script/build
+
+upload:
+	ifndef NAMESPACE
+	$(error NAMESPACE is not set)
+	endif
```

Comment:
This doesn't seem to accept a NAMESPACE:
```
$ make upload
Makefile:12: *** NAMESPACE is not set.  Stop.

$ make upload NAMESPACE=cynerva
Makefile:12: *** NAMESPACE is not set.  Stop.

$ NAMESPACE=cynerva make upload
Makefile:12: *** NAMESPACE is not set.  Stop.
```

I think maybe NAMESPACE needs to be defined as a variable at the top of the Makefile? It sounds like `ifndef` should still work here as long as the variable is empty:
```
ifndef variable-name
    If the variable variable-name has an empty value, the text-if-true is effective; otherwise, the text-if-false, if any, is effective.
```